extension/llm/runner/multimodal_runner.cpp

```diff
@@ -0,0 +1,215 @@
+    stats_->model_load_end_ms = time_in_ms();
+  }
+
+  if (config.warming) {
+    ET_LOG(Info, "Doing a warmup run...");
+  }
+
+  RUNNER_ET_LOG(
+      config.warming,
+      "RSS after loading model: %f MiB (0 if unsupported)",
+      get_rss_bytes() / 1024.0 / 1024.0);
+
+  // Wrap the token_callback with print function
+  std::function<void(const std::string&)> wrapped_callback =
+      [token_callback, config](const std::string& piece) {
+        if (!config.warming) {
+          safe_printf(piece.c_str());
+          fflush(stdout);
+        }
+        if (token_callback) {
+          token_callback(piece);
+        }
+      };
+
+  // Reset internal state and start inference
+  pos_ = 0;
+  stats_->inference_start_ms = time_in_ms();
+
+  uint64_t prefill_next_token = 0;
+  // Process multimodal inputs in order
+  for (const auto& input : inputs) {
+    prefill_next_token = ET_UNWRAP(prefill_input(input, 0, 0));
+  }
+
+  stats_->first_token_ms = time_in_ms();
+  stats_->prompt_eval_end_ms = time_in_ms();
+  stats_->num_prompt_tokens = pos_;
+
+  wrapped_callback(ET_UNWRAP_TOKENIZER(
+      tokenizer_->decode(prefill_next_token, prefill_next_token)));
+
+  RUNNER_ET_LOG(
+      config.warming,
+      "RSS after multimodal input processing: %f MiB (0 if unsupported)",
+      get_rss_bytes() / 1024.0 / 1024.0);
+
+  // Resolve max_new_tokens based on config
+  int64_t max_context_len =
+      metadata_.at(kMaxContextLen) - 0; // No start_pos offset
+  int32_t max_new_tokens = config.resolve_max_new_tokens(max_context_len, pos_);
+
+  ET_LOG(
+      Info,
+      "Max new tokens resolved: %d, pos_ %" PRId64 ", max_context_len %" PRId64,
+      max_new_tokens,
+      pos_,
+      max_context_len);
+
+  ET_CHECK_OR_RETURN_ERROR(
+      max_new_tokens > 0,
+      InvalidArgument,
+      "Max new tokens %d is less than or equal to 0",
+      max_new_tokens);
+
+  // Generate tokens using the text token generator
+  std::vector<uint64_t> prompt_tokens = {prefill_next_token};
+  int64_t num_generated_tokens = ET_UNWRAP(text_token_generator_->generate(
+      /*tokens=*/prompt_tokens,
+      /*start_pos=*/pos_,
+      /*max_new_tokens=*/max_new_tokens -
+          1, // Subtract 1 because prefill already generated 1 token
+      /*temperature=*/config.temperature,
+      /*token_callback=*/wrapped_callback));
+
+  pos_ += num_generated_tokens;
+  // Update stats
+  stats_->num_generated_tokens = num_generated_tokens;
+  // Finalize stats and call callback
+  stats_->inference_end_ms = time_in_ms();
+  if (!config.warming) {
+    printf("\n");
+  }
+
+  if (config.warming) {
+    ET_LOG(Info, "Warmup run finished!");
+  } else {
+    // Do not print report during warmup
+    print_report(*stats_);
+  }
+
+  if (stats_callback) {
+    stats_callback(*stats_);
+  }
+
+  return Error::Ok;
+}
+
+Result<uint64_t> MultimodalRunner::prefill_input(
+    const MultimodalInput& input,
+    int8_t bos,
+    int8_t eos) {
+  if (input.is_text()) {
```

Comment:
You can simplify this code block, so there are only if (no else if and else) since every branch returns

Reply:
Added image, text and else so will need `else if` for sure.